### PR TITLE
Issue #58812 : DOC : Added doc string to announce.py and contibutors.py

### LINKED ISSUE #58812: DOC: Docstrings missing from .py files in Sphinxext docs folder
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://github.com/pandas-dev/pandas/blob/main/doc/sphinxext/contributors.py
https://github.com/pandas-dev/pandas/blob/main/doc/sphinxext/announce.py

### Documentation problem

Both announce.py and contributors.py are missing the class and function docstrings encouraged by Pandas documentation. There are around 5 functions and 1 class that need documenting. Docstrings are always helpful for new users looking to contribute to the codebase.

### Suggested fix for documentation

The docs have a detailed docstring how-to guide:
https://pandas.pydata.org/docs/dev/development/contributing_docstring.html

### PR DESCRIPTION
**Pull request by malhar2460**
- [x] closes #58812  (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by Aloqeely — COMMENTED** [review]
Amazing work! I think this is a bit overkill though, in my opinion only the summary should be kept because this is a command line script so we don't really need to explain how to use these functions. Or maybe don't include the examples section which takes a lot of space

File: doc/sphinxext/announce.py [hunk 1]
```diff
@@ -57,6 +57,34 @@
 
 
 def get_authors(revision_range):
+    """
+    Extract authors from a range of revisions
```
[the remark this answers is not in the document]
Comment (malhar2460) [reply doc/sphinxext/announce.py:61]: Done 👍

**Comment by malhar2460** [comment]
Okay so should i remove the long summary or examples from doc string

**Comment by jbrockmendel** [comment]
iirc this file is vendored so we wouldn’t want to edit it

**Comment by malhar2460** [comment]
@jbrockmendel Sorry im new to this, can you give me some more detail about what to do, i didn't understand

**Comment by Aloqeely** [comment]
> iirc this file is vendored so we wouldn’t want to edit it

Ah, I didn't know that. After further search it seems most of the copied code is from scipy `tools/authors.py` which has "This script is in the public domain." written at the top of the file, so I think it's ok to edit it

**Comment by malhar2460** [comment]
@Aloqeely I have removed the examples from doc string and added full stop to the shot summary, is there any other change that needs to be made?

**Comment by Aloqeely** [comment]
@malhar2460 I am sorry but as mentioned here, this file is vendored so it isn't touched very often and I don't think we're gonna update it just to add the docstrings. Really sorry for taking much of your time!

On a note I'm still OK with this going through since the functions here are different from the ones upstream, but leaving final decision for @jbrockmendel

**Comment by malhar2460** [comment]
Okay, np👍🏻

**Pull request closed without merging**